erdpy/cli_delagation.py

```diff
@@ -0,0 +1,251 @@
+                                           "Change service fee must be called by the contract owner")
+    sub.add_argument("--service-fee", required=True, help="new service fee value")
+    sub.add_argument("--delegation-contract", required=True, help="address of the delegation contract")
+    _add_common_arguments(sub)
+    sub.set_defaults(func=change_service_fee)
+
+    # modify total delegation cap
+    sub = cli_shared.add_command_subparser(subparsers, "staking-provider", "modify-delegation-cap",
+                                           "Modify delegation cap must be called by the contract owner")
+    sub.add_argument("--delegation-cap", required=True, help="new delegation contract capacity")
+    sub.add_argument("--delegation-contract", required=True, help="address of the delegation contract")
+    _add_common_arguments(sub)
+    sub.set_defaults(func=modify_delegation_cap)
+
+    # set automatic activation
+    sub = cli_shared.add_command_subparser(subparsers, "staking-provider", "automatic-activation",
+                                           "Automatic activation must be called by the contract owner")
+
+    sub.add_argument("--set", action="store_true", required=not (utils.is_arg_present("--unset", sys.argv)),
+                     help="set automatic activation True")
+    sub.add_argument("--unset", required=not (utils.is_arg_present("--set", sys.argv)),
+                     help="set automatic activation False")
+
+    sub.add_argument("--delegation-contract", required=True, help="address of the delegation contract")
+    _add_common_arguments(sub)
+    sub.set_defaults(func=automatic_activation)
+
+
+def _add_common_arguments(sub: Any):
+    cli_shared.add_proxy_arg(sub)
+    cli_shared.add_wallet_args(sub)
+    cli_shared.add_tx_args(sub, with_receiver=False, with_data=False, with_estimate_gas=True)
+    cli_shared.add_broadcast_args(sub, relay=False)
+    cli_shared.add_outfile_arg(sub, what="signed transaction, hash")
+
+
+def do_create_delegation_contract(args: Any):
+    cli_shared.check_broadcast_args(args)
+    cli_shared.prepare_nonce_in_args(args)
+    staking_provider.prepare_args_for_create_new_staking_contract(args)
+    tx = do_prepare_transaction(args)
+
+    try:
+        cli_shared.send_or_simulate(tx, args)
+    finally:
+        tx.dump_to(args.outfile)
+
+
+def get_contract_address_by_deploy_tx_hash(args: Any):
+    args = utils.as_object(args)
+    omit_fields = cli_shared.parse_omit_fields_arg(args)
+
+    proxy = ElrondProxy(args.proxy)
+
+    transaction = proxy.get_transaction(args.create_tx_hash, with_results=True)
+    utils.omit_fields(transaction, omit_fields)
+    utils.get_sc_address_from_tx(transaction)
+
+
+def add_new_nodes(args: Any):
+    cli_shared.check_broadcast_args(args)
+    cli_shared.prepare_nonce_in_args(args)
+    staking_provider.prepare_args_for_add_nodes(args)
+    tx = do_prepare_transaction(args)
+
+    try:
+        cli_shared.send_or_simulate(tx, args)
```

Comment:
why do we call `send_or_simulate` function and not calling `send`?

Reply:
because for this command we also have the option simulate.